tests/Doctrine/Tests/DBAL/Functional/DataAccessTest.php

```diff
@@ -863,6 +863,49 @@ private function skipOci8AndMysqli()
             $this->markTestSkipped('Mysqli driver dont support this feature.');
         }
     }
+
+    public function testIteratorCanBeIteratedMultipleTimes()
+    {
+        $skippedMessage = <<<'MSG'
+Iterators are not rewindable for either PDO or Non PDO drivers. This is because PDO by default is not rewindable and 
+to be consistent we have not implemented rewindable iterators for non PDO drivers. Should PDO ever become rewindable by
+default we should wrap the legacy drivers in Doctrine\DBAL\RewindableGenerator for example on the Mysqli driver you 
+would do something like the following:
+
+return new RewindableGenerator(function() {
+    $this->_stmt->data_seek(0);
+    while ($row = $this->fetch()) {
+        yield $row;
+    }
+});
+
+If this has been done you can include this test. 
+MSG;
+
+        $this->markTestSkipped($skippedMessage);
```

Comment:
Sorry, not sure I totally understand. 

I think you're suggesting to decorate the statement with a spy of some kind so that we can examine that `fetch()` is being called as we'd expect? 

If that is correct, I'm still not 100% sure how to implement that. Statements, as far as I can tell, [are new'd up in the relevant connection class](https://github.com/doctrine/dbal/blob/master/lib/Doctrine/DBAL/Driver/Mysqli/MysqliConnection.php#L120), meaning that we'd have to mock the connection or rewrite to allow a statement to be injected. Is this the path you were thinking of or am I way off?